gradle.lockfile

```diff
@@ -0,0 +1,206 @@
+# This is a Gradle generated file for dependency locking.
```

Comment:
What problem is this fixing? I'd lean towards not adding this extra complexity, if there's nothing to fix...

Reply:
2 reasons I added this for, open to discuss and remove if the value add does not outweigh the added complexity

- Versions of dependencies matter here: incompatible versions might cause errors during compile / runtime. Some of the libraries did not specify the version to be included, so pulling the latest version might lead to breaking changes. Hence added this to avoid them. Example: some libraries in the build file refer to older library versions due to compatibility errors 
- Keeping it consistent with other newer repos (like WSM).. this repo is relatively older and does not align to the newer standards followed. Consistent structure eases maintenance 

(Will sync up when rest pr if ready - these changes are easy to remove)